tests/Testable/NavigationTests.elm

```diff
@@ -0,0 +1,245 @@
+                                )
+                , test "goes forward, asking to trigger location msg" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (New "/bar")
+                            |> thenUpdate (Jump -2)
+                            |> thenUpdate (Jump 1)
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    , getLocation "https://elm.testable/bar"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                , test "has a limit to jumping back" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (Jump -100)
+                            |> Expect.equal
+                                ( ( 0
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg initialLocation
+                                )
+                , test "has a limit to jumping forward" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (Jump -1)
+                            |> thenUpdate (Jump 100)
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                ]
+            , describe "visit url"
+                [ test "visit adds a new url to the history, asking to trigger location msg" <|
+                    \() ->
+                        init
+                            |> update (Visit "/foo")
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                ]
+            ]
```

Comment:
Yes, like I mentioned above, those are more unitary and the other more integrated, I believe both are important, here we catch the edge cases (and there are a lot for urls), while there we check that everything is working well together (also the helpers functions that only live on TestContext, like `|> navigate "/foo"`)